tests/lib/nested.sh

```diff
@@ -613,6 +613,31 @@ nested_configure_cloud_init_on_core20_vm() {
     kpartx -d "$IMAGE"
 }
 
+nested_save_serial_log() {
+    if [ -f "${NESTED_LOGS_DIR}/serial.log" ]; then
+        for i in $(seq 50); do
+            if [ ! -f "${NESTED_LOGS_DIR}/serial.log.${i}" ]; then
+                cp "${NESTED_LOGS_DIR}/serial.log" "${NESTED_LOGS_DIR}/serial.log.${i}"
+                break
+            fi
+        done
+        # make sure we start with clean log file
+        echo > "${NESTED_LOGS_DIR}/serial.log"
+    fi
+}
+
+nested_print_serial_log() {
+    if [ -f "${NESTED_LOGS_DIR}/serial.log" ]; then
+        # here we disable SC2045 because previously it is checked there is at least
+        # 1 file which matches. In this case ls command is needed because it is important
+        # to get the list in reverse order.
+        # shellcheck disable=SC2045
+        for logfile in $(ls -r "${NESTED_LOGS_DIR}"/serial.log*); do
```

Comment:
If we have more than 10 files, the order will be somewhat confusing. From little testing this is the order we get with `ls -r`:

```
serial.log.9
serial.log.8
serial.log.7
serial.log.6
serial.log.5
serial.log.4
serial.log.3
serial.log.2
serial.log.10
serial.log.1
serial.log
```

I think we should use `seq 9` in `nested_save_serial_log`, so that we don't run into this problem. Also note, that the most recent log will have the highest number, so the order is still confusing. Maybe we should do something like this to ensure that the logs are printed in the order from the oldest to most recent?

```
for logfile in "$(ls "$NESTED_LOGS_DIR"/serial.log.*)"; do
    cat "$logfile"
done
cat "$NESTED_LOGS_DIR"/serial.log || true
```

Reply:
Sure, thanks for the suggestion